tests/backup/backup_resiliency_test.go

```diff
@@ -758,3 +761,240 @@ var _ = Describe("{CancelAllRunningBackupJobs}", func() {
+				}(backupName)
+			}
+			wg.Wait()
+			log.Infof("The list of restores are: %v", restoreNames)
+		})
+		Step("Scaling MongoDB statefulset replica to 0 and back to original replica while restore is in progress", func() {
+			log.InfoD("Scaling MongoDB statefulset replica to 0 while restore is in progress")
+			*statefulSet.Spec.Replicas = scaledDownReplica
+			statefulSet, err = apps.Instance().UpdateStatefulSet(statefulSet)
+			log.Infof("mongodb replica after scaling to 0 is %v", *statefulSet.Spec.Replicas)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Scaling down MongoDB statefulset replica to 0"))
+			dash.VerifyFatal(*statefulSet.Spec.Replicas == scaledDownReplica, true, "Getting mongodb statefulset replica after scaling down")
+			log.InfoD("Sleeping for 1 minute so that at least one request is hit to mongodb for the restores taken")
+			time.Sleep(1 * time.Minute)
+			log.InfoD("Scaling MongoDB statefulset to original replica while restore is in progress")
+			statefulSet, err = apps.Instance().GetStatefulSet(mongodbStatefulset, pxBackupNS)
+			*statefulSet.Spec.Replicas = originalReplicaCount
+			statefulSet, err = apps.Instance().UpdateStatefulSet(statefulSet)
+			log.Infof("mongodb replica after scaling back to original replica is %v", *statefulSet.Spec.Replicas)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Scaling MongoDB statefulset back to original replica"))
+			dash.VerifyFatal(*statefulSet.Spec.Replicas == originalReplicaCount, true, "Verify mongodb statefulset replica after scaling back to original")
+			log.Infof("Verify that at least one mongodb pod is in Ready state")
+			mongoDBPodStatus := func() (interface{}, bool, error) {
+				statefulSet, err = apps.Instance().GetStatefulSet(mongodbStatefulset, pxBackupNS)
+				if err != nil {
+					return "", true, err
+				}
+				if statefulSet.Status.ReadyReplicas < 1 {
+					return "", true, fmt.Errorf("no mongodb pods are ready yet")
+				}
+				return "", false, nil
+			}
+			_, err = task.DoRetryWithTimeout(mongoDBPodStatus, 5*time.Minute, 30*time.Second)
+			log.Infof("Number of mongodb pods in Ready state are %v", statefulSet.Status.ReadyReplicas)
+			dash.VerifyFatal(statefulSet.Status.ReadyReplicas > 0, true, "Verifying that at least one mongodb pod is in Ready state")
+		})
+		Step("Check if restore is successful after MongoDB statefulset is scaled back to original replica", func() {
+			log.InfoD("Check if restore is successful after MongoDB statefulset is scaled back to original replica")
+			ctx, err = backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, restoreName := range restoreNames {
+				restoreStatus, err := restoreSuccessCheck(restoreName, orgID, 0, 0, ctx)
+				dash.VerifyFatal(err, nil, "Getting the status for restore-"+restoreName)
+				dash.VerifyFatal(restoreStatus, true, "Verifying the restore status for restore-"+restoreName)
+			}
+		})
+	})
+
+	JustAfterEach(func() {
```

Comment:
We need a safety check here. Lets say something fails before you can scale up mongo to original replica count, then all the subsequent tests will fail. So you have to do something like this

```
statefulSet, err = apps.Instance().GetStatefulSet(mongodbStatefulset, pxBackupNS)
if *statefulSet.Spec.Replicas != originalReplicaCount{
    *statefulSet.Spec.Replicas = originalReplicaCount
     statefulSet, err = apps.Instance().UpdateStatefulSet(statefulSet)
// wait till replica set becomes equal to original replica set
}
```

Reply:
done

Reply:
To avoid ambiguity, would it be feasible to rename statefulSet from UpdateStatefulSet to something like updatedStatefulSet?

Reply:
I meant the output "statefulSet" in "statefulSet, err = apps.Instance().UpdateStatefulSet(statefulSet)"